### PR TITLE
jsdoc: document transform methods

### PR DESCRIPTION
**Pull request by machikoyasuda**
## What this PR does
1. Adds a `@module` called `cartOrderTransform` to the sidebar and a page.
2. Cleans up the way these methods show up on that page.
3. Adds a blurb about how to use transform methods

## How that renders

The sidebar and page:
screen shot 2017-10-18 at 11 09 29 am

A deprecated method:
screen shot 2017-10-18 at 11 10 49 am

### CODE REVIEW

File: lib/collections/transform/cartOrder.js [hunk 1]
```diff
@@ -46,26 +47,23 @@ function getSummary(items, prop, prop2, shopId) {
...
- * Reaction transform collections
- *
- * transform methods used to return cart and order calculated values
- * count, subTotal, shipping, taxes, total
- * are calculated by a transformation on the collection
- * and are available to use in template as cart.xxx or order.xxx
- * in template: {{cart.getCount}}
- * in code: Cart.findOne().getTotal()
+ * Reaction transform methods on Collections
+ * @file Use transform methods to return Cart and Order calculated values: count, subTotal, shipping, taxes, total. Use these methods on Cart and Orders in templates, `{{cart.getCount}}` and also in code, `Cart.findOne().getTotal()`. These are calculated by using a Mongo Collection's {@link http://docs.meteor.com/api/collections.html#Mongo-Collection transform functionality}.
```
Comment (machikoyasuda) [inline lib/collections/transform/cartOrder.js]: @zenweasel Is this a good blurb for the `@file`? It shows up at the top of the page. Edits encouraged!

File: lib/collections/transform/cartOrder.js [hunk 2]
```diff
@@ -149,7 +145,8 @@ export const cartOrderTransform = {
     }, {});
   },
   /**
-   * Discount for cart/order. Grabs discounts from the invoice records if they exist, otherwise from this.discounts
+   * @summary Discount for cart/order. Grabs discounts from the invoice records if they exist, otherwise from this.discounts
+   * @example const cartTaxesByShop = cart.getTaxesByShop();
    * @method orderDiscounts
```
Comment (zenweasel) [inline lib/collections/transform/cartOrder.js]: Looks like this method name needs to be updated? (error in the original code)

File: lib/collections/transform/cartOrder.js (hunk 1 above)
Comment (zenweasel) [reply lib/collections/transform/cartOrder.js]: Can we format this to fit within the 120 line width? I can't read all of it here as it flows off the page

Comment (aaronjudd) [reply lib/collections/transform/cartOrder.js]: @machikoyasuda needs to follow our standard code guidelines.

**Comment by aaronjudd** [comment]
@zenweasel @machikoyasuda shouldn't eslint be failing for more than 120 chars?

**Comment by jshimko** [comment]
Nope.  Looks like we changed it to 160 at some point.

https://github.com/reactioncommerce/reaction/blob/master/.eslintrc#L180-L183

**Comment by jshimko** [comment]
Looks like it was 0.15.1 in #1314

https://github.com/reactioncommerce/reaction/blame/master/.eslintrc#L180

**Comment by aaronjudd** [comment]
I know we used to have it at 160, that's over a year ago though, I would have sworn that @zenweasel pushed through / convinced us to switch to 120, although does appear we've been steady at 160-> so my bad...

**Comment by zenweasel** [comment]
@aaronjudd @machikoyasuda I did but my bad for just changing `editorconfig` and not `eslintrc`.

File: lib/collections/transform/cartOrder.js [hunk 3]
```diff
@@ -46,27 +47,26 @@ function getSummary(items, prop, prop2, shopId) {
...
- * transform methods used to return cart and order calculated values
- * count, subTotal, shipping, taxes, total
- * are calculated by a transformation on the collection
- * and are available to use in template as cart.xxx or order.xxx
- * in template: {{cart.getCount}}
- * in code: Cart.findOne().getTotal()
+ * Reaction transform methods on Collections
+ * @file Use transform methods to return Cart and Order calculated values: count, subTotal, shipping, taxes, total.
+ * Usethese methods on Cart and Orders in templates, `{{cart.getCount}}` and in code, `Cart.findOne().getTotal()`.
+ * These use Mongo Collection {@link http://docs.meteor.com/api/collections.html#Mongo-Collection transforms}.
```
Comment (zenweasel) [inline lib/collections/transform/cartOrder.js]: You mean Meteor collections not Mongo collections

File: lib/collections/transform/cartOrder.js [hunk 4]
```diff
@@ -46,27 +47,26 @@ function getSummary(items, prop, prop2, shopId) {
...
- *
- * transform methods used to return cart and order calculated values
- * count, subTotal, shipping, taxes, total
- * are calculated by a transformation on the collection
- * and are available to use in template as cart.xxx or order.xxx
- * in template: {{cart.getCount}}
- * in code: Cart.findOne().getTotal()
+ * Reaction transform methods on Collections
+ * @file Use transform methods to return Cart and Order calculated values: count, subTotal, shipping, taxes, total.
+ * Usethese methods on Cart and Orders in templates, `{{cart.getCount}}` and in code, `Cart.findOne().getTotal()`.
```
Comment (zenweasel) [inline lib/collections/transform/cartOrder.js]: typo

**Pull request merged**